orttraining/orttraining/training_api/onnxruntime_training_c_api.cc

```diff
@@ -21,6 +35,7 @@ ORT_API_STATUS_IMPL(OrtApis::CreateTrainingSession, _In_ const OrtEnv* env, _In_
     train_sess = std::make_unique<onnxruntime::training::api::TrainingSession>(
         env->GetEnvironment(),
         options == nullptr ? onnxruntime::SessionOptions() : options->value,
+        CreateProviders(options->provider_factories),
```

Comment:
needs check for nullptr... I think we touched in this in my pr review... we need a check for options because options OrtSessionOptions is allowed be nullptr... it just means user wants to go with default options... 
should be changed to something like : options == nullptr ? {} : CreateProviders(...)